drivers/sensor/ti/ina23x/ina230.c

```diff
@@ -185,14 +231,19 @@ static int ina230_calibrate(const struct device *dev)
 static int ina230_init(const struct device *dev)
 {
 	const struct ina230_config *const config = dev->config;
+	uint16_t config_raw = config->config;
 	int ret;
 
 	if (!device_is_ready(config->bus.bus)) {
 		LOG_ERR("I2C bus %s is not ready", config->bus.bus->name);
 		return -ENODEV;
 	}
 
-	ret = ina23x_reg_write(&config->bus, INA230_REG_CONFIG, config->config);
+	if (ina230_is_high_precision(config)) {
+		config_raw |= INA236_CONFIG_ADC_RANGE;
```

Comment:
The `INA236_CONFIG_ADC_RANGE` defines only INA236. I think it would be better to rename the `ina230_is_high_precision` function to `ina23x_is_high_precision` instead.